### PR TITLE
Required Virtual Template Parameters

### PR DESCRIPTION
**Pull request by jntullo**
Minor update to the amazon virtual template to account for required parameters

### CODE REVIEW

**Comment by jntullo** [comment]
@durandom realized this breaks a test currently in MIQ. Going to do a PR to update that conflict, hold off on this for now.

**Comment by jntullo** [comment]
related PR manageiq/manageiq#9555

File: app/models/manageiq/providers/amazon/cloud_manager/virtual_template.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 class ManageIQ::Providers::Amazon::CloudManager::VirtualTemplate < ::ManageIQ::Providers::CloudManager::VirtualTemplate
-  validates :cloud_subnet_id, :availability_zone_id, :ems_ref, :presence => true
+  validates :cloud_network_id, :availability_zone_id,
+            :ems_ref, :ems_id, :flavor_id, presence: true
```
Comment (durandom) [inline app/models/manageiq/providers/amazon/cloud_manager/virtual_template.rb]: _unfortunately_ we prefer hash rockets
`:presence => true`

@Fryguy can we get miq-bot to comment on PRs on this repo?

File: app/models/manageiq/providers/amazon/cloud_manager/virtual_template.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 class ManageIQ::Providers::Amazon::CloudManager::VirtualTemplate < ::ManageIQ::Providers::CloudManager::VirtualTemplate
-  validates :cloud_subnet_id, :availability_zone_id, :ems_ref, :presence => true
+  validates :cloud_network_id, :availability_zone_id,
```
Comment (durandom) [inline app/models/manageiq/providers/amazon/cloud_manager/virtual_template.rb]: i think validating presence of the actual object vs the id is better
see [this post](http://railsguides.net/belongs-to-and-presence-validation-rule1/) for a good explanation

File: app/models/manageiq/providers/amazon/cloud_manager/virtual_template.rb (hunk 1 above)
Comment (jntullo) [inline app/models/manageiq/providers/amazon/cloud_manager/virtual_template.rb]: oh boo hash rockets. I had the hash rocket, but local rubocop changed it and I got excited that this repo wasn't going to use them hah but I see there's no `rubocop.yml` so it makes sense now.

File: app/models/manageiq/providers/amazon/cloud_manager/virtual_template.rb (hunk 2 above)
Comment (jntullo) [inline app/models/manageiq/providers/amazon/cloud_manager/virtual_template.rb]: yes that makes more sense. will do, thanks @durandom!

**Pull request merged**